src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -3135,6 +3135,99 @@ private async Task<(long Index, object Result)> SendToNodeAsync(TransactionOpera
             return (command.Result.RaftCommandIndex, command.Result.Data);
         }
 
+        public async Task WaitForExecutionOnSpecificNode(JsonOperationContext context, string nodeTag, long index)
+        {
+            await WaitForExecutionOnRelevantNodesAsync(context, members: new List<string> { nodeTag }, index);
+        }
+
+        public async Task WaitForExecutionOnRelevantNodesAsync(JsonOperationContext context, List<string> members, long index)
+        {
+            await Cluster.WaitForIndexNotification(index); // first let see if we commit this in the leader
+
+            if (members.Count == 0)
+                throw new InvalidOperationException("Cannot wait for execution when there are no nodes to execute ON.");
+
+            var executors = GetClusterRequestExecutorsForMembers(members);
+
+            try
+            {
+                using (var cts = CancellationTokenSource.CreateLinkedTokenSource(ServerShutdown))
+                {
+                    cts.CancelAfter(Configuration.Cluster.OperationTimeout.AsTimeSpan);
```

Comment:
Since we decided not to transfer this method into `RequestHandler`, we cannot access the `CreateHttpRequestBoundTimeLimitedOperationToken` method. I have corrected it by using the `OperationCancelToken` class.